velox/core/PlanNode.cpp

```diff
@@ -152,6 +190,96 @@ void AggregationNode::addDetails(std::stringstream& stream) const {
   }
 }
 
+// static
+const char* AggregationNode::stepName(AggregationNode::Step step) {
+  switch (step) {
+    case Step::kPartial:
+      return "PARTIAL";
+    case Step::kFinal:
+      return "FINAL";
+    case Step::kIntermediate:
+      return "INTERMEDIATE";
+    case Step::kSingle:
+      return "SINGLE";
+    default:
+      VELOX_UNREACHABLE();
+  }
+}
+
+// static
+AggregationNode::Step AggregationNode::stepFromName(const std::string& name) {
+  static const std::unordered_map<std::string, AggregationNode::Step> kSteps = {
+      {"PARTIAL", Step::kPartial},
```

Comment:
not 100% if it would work, but could we make the function above contexpr and call it from here, so we don't need to repeat each string twice?

```
  {stepName{Step::kPartial}, Step::kPartial}, ...
```

Reply:
Here is what I did (applies to all enums). Let me know if this works better.

```
namespace {
std::unordered_map<std::string, AggregationNode::Step> nameToStep() {
  return {
      {"PARTIAL", AggregationNode::Step::kPartial},
      {"FINAL", AggregationNode::Step::kFinal},
      {"INTERMEDIATE", AggregationNode::Step::kIntermediate},
      {"SINGLE", AggregationNode::Step::kSingle},
  };
}

template <typename K, typename V>
std::unordered_map<V, K> invert(const std::unordered_map<K, V>& mapping) {
  std::unordered_map<V, K> inverted;
  for (const auto& [key, value] : mapping) {
    inverted.emplace(value, key);
  }
  return inverted;
}
} // namespace

// static
const char* AggregationNode::stepName(AggregationNode::Step step) {
  static const auto kSteps = invert(nameToStep());
  return kSteps.at(step).c_str();
}

// static
AggregationNode::Step AggregationNode::stepFromName(const std::string& name) {
  static const auto kSteps = nameToStep();
  return kSteps.at(name);
}
```